pinot-core/src/main/java/org/apache/pinot/core/periodictask/PeriodicTaskScheduler.java

```diff
@@ -108,4 +111,59 @@ public synchronized void stop() {
       _tasksWithValidInterval.parallelStream().forEach(PeriodicTask::stop);
     }
   }
+
+  /** @return true if task with given name exists; otherwise, false. */
+  public boolean hasTask(String periodicTaskName) {
+    for (PeriodicTask task : _tasksWithValidInterval) {
+      if (task.getTaskName().equals(periodicTaskName)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /** @return List of tasks name that will run periodically. */
+  public List<String> getTaskNameList() {
+    List<String> taskNameList = new ArrayList<>();
```

Comment:
why create a new list? why not just return a `Collections.unmodifiableList` of this same one?

Reply:
`_tasksWithValidInterval` is a `List<PeriodicTask>`, but here we are returning a `List<String>`.

Reply:
ah. lambda wil be nice.